project/Build.scala

```diff
@@ -108,6 +108,19 @@ object JobServerBuild extends Build {
     dockerfile in docker := {
       val artifact = (assemblyOutputPath in assembly in jobServerExtras).value
       val artifactTargetPath = s"/app/${artifact.name}"
+
+      val sparkBuild = s"spark-$sparkVersion"
+      val sparkBuildCmd = scalaBinaryVersion.value match {
+        case "2.10" =>
+          "./make-distribution.sh -Phadoop-2.4 -Phive"
```

Comment:
The current docker image is also fixed to Hadoop 2.4 and I didn't want to change everything in one PR. I suggest trying this PR out and see if it fits the build process, then expand on it afterwards (like configuring Hadoop and Java, building SparkR, ...).